org.eclipse.draw2d/src/org/eclipse/draw2d/FigureCanvas.java

```diff
@@ -462,6 +463,26 @@ private void scrollChildren(int dx, int dy, Rectangle blit, Point dest) {
 		}
 	}
 
+	private void scrollRedraw(int dx, int dy, Rectangle clientArea, Rectangle blit) {
+		Rectangle expose = clientArea.getCopy();
+		if (dx != 0) {
+			expose.width = Math.abs(dx);
+			if (dx < 0) { // Moving left?
+				expose.x += blit.width; // Move expose area right
```

Comment:
This used to be `expose.x = dest.x + blit.width`, but given that `dest = clientArea.getTopLeft()` and `expose = clientArea.copy()`, I assume this section is identical to the current implementation?